03-MVC-AngularJS/module.js

```diff
@@ -0,0 +1,113 @@
+var myApp = angular.module('myApp', []);
+
+myApp.factory('Storage', function () {
+	return {
+		get: function(string) {
+			var json_obj = localStorage.getItem(string);
+			return JSON.parse(json_obj);
+		},
+		post: function(list, string) {
+			var json_obj = JSON.stringify(list);
+			localStorage.setItem(string, json_obj);
+		}
+	};
+});
+ 
+myApp.service('Storage_Serv', function(Storage){
+	this.get = function(string) {
+		return Storage.get(string);
+	}
+	this.post = function(list, string){
+		Storage.post(list, string);
+	}
+});
+ 
+myApp.controller('MainCtrl', function($scope, Storage_Serv) {
+
+	//Storage_Serv.post(movie, 'movies');
+
+	$scope.movie_list = Storage_Serv.get('movies');
+
+	$scope.hide_details = true;
+	$scope.selected_movie;
+
+	$scope.show_details = function(movie) {
+		$scope.hide_details = $scope.change_visibility(movie);
+	};
+
+	$scope.get_selected_movie = function() {
+		return $scope.selected_movie;
+	};
+
+	$scope.change_visibility = function(movie) {
+		
+		if($scope.selected_movie == movie && !$scope.hide_details) 
+			$scope.hide_details = true;
+		else 
+		{
+			$scope.selected_movie = movie;
+			$scope.hide_details = false;
+		}
+		
+		return $scope.hide_details;
+	};
+
+	$scope.add_movie = function(){
+		if (document.getElementById('in_movie_title').value != '' && document.getElementById('in_movie_year').value != '' && document.getElementById('in_movie_duration').value != '') {
```

Comment:
it's wrong to use document.getElementById inside angular, you should **only** use it's own way to interact with UI. In this framework you have the ng-model with two way data binding when dealing with forms where user interacts with.

If you attach an ng-model to an input, when it is changed it's instantaneously updated in the model. So you only have to access to the model here in order to get the selected / input values instead of getting them from DOM.